src/atom_finder/classifier/operator-precedence.clj

```diff
@@ -32,92 +32,106 @@
+    [:bitwise_not :de_incr] [:arith_unary :bitwise_bin] [:not :bitwise_bin] [:bitwise_bin :cond]
+    [:bitwise_not :bitwise_bin]})
 
 (defn confusing-operator-combination?
   "Is this combination of the operator groups confusing?"
   [combination]
-  (or (specific-confusing-operator-combination? combination)
-      (some (partial contains? always-confusing-operator-groups) combination)))
+  (or (order-sensitive-opt-combination combination)
+      (order-insensitive-opt-combination (sort combination))
+
+      ;;SPECIAL CASE 1: One of the entry is comma operator;
+      (some (partial = :comma) combination)))
 
 (defn operator-group-pair?
   [node]
   (and (operator-group node)
        (some operator-group (children node))))
 
-(defn group-pair
-  "returns a collection of operator group pairs between the node and its children
-  , if a second parameter is passed, use that as the collection instead of the children"
-  ([node]
-   (group-pair node (children node)))
-
-  ([node collection]
-   (->> collection
-        (map operator-group)
-        (remove nil?)
-        (map (fn [child-group] (sort [(operator-group node) child-group]))))))
-
-;;====================================
-;;For binary operator special case
-;;====================================
 (def rvalue-unary-ops?
-  "Is this a unary operator that can operate on an rvalue? (! - + *)"
-   (comp #{:arith_unary :not :pointer} operator-group))
+  "HELPER:Is this a unary operator that can operate on an rvalue? (! - + *)"
+   (comp #{:arith_unary :not :pointer :bitwise_not} operator-group))
 
 (defn unparenthesized-unary-in-children
-  "Is this node not a bracket and constains unary operator in its children?"
+  "HELPER:Is this node not a bracket and constains unary operator in its children?"
   [node]
   (if (not (paren-node? node))
-      (filter-tree #(rvalue-unary-ops? %) node)))
+      (filter #(rvalue-unary-ops? %) (children node))))
 
-(defn group-pairs-in-binary-ops
-  "Special function for returning a collection of operator group in binary operator's children, binary operator should ignore some groups of operators in its second operand"
+(defn unary-before-binary-pairs
+  "Special function for returning a collection of unary operator group in the first operand of the binary operator node"
   [node]
-  (->> (concat [(get-in-tree [0] node)
-                (if (not (rvalue-unary-ops? (get-in-tree [1] node)))
-                  (get-in-tree [1] node) nil)]
-
-               (unparenthesized-unary-in-children (get-in-tree [0] node)))
+  (->> (unparenthesized-unary-in-children (get-in-tree [0] node))
        (map operator-group)
        (remove nil?)
-       (map (fn [child-group] (sort [(operator-group node) child-group])))))
-;;====================================
-;;
+       (map (fn [child-group] [child-group (operator-group node)]))))
 
+(defn map-group-pair
+  "map the group-pair based on the type of node"
```

Comment:
I don't understand what this function does. Can you add some more description to the docstring?